core/user_api.php

```diff
@@ -305,9 +305,67 @@ function user_ensure_name_unique( $p_username, $p_user_id = null ) {
 	}
 }
 
+/**
+ * Returns a list of duplicate email addresses (case-insensitive).
+ *
+ * Each array entry's key is the e-mail address, and the value is an array
+ * listing the associated user accounts, with key=id and value=username.
+ *
+ * Note that the complex query used by this function is quite expensive (average
+ * execution time of about 0.3 seconds for 50K users with 1000 duplicates).
+ *
+ * @return array
+ */
+function user_get_duplicate_emails() {
+	static $s_duplicate_emails;
+
+	if( $s_duplicate_emails === null ) {
+		$t_user_table = db_get_table( 'user' );
+
+		# Using a sub-query within the IN clause's SELECT statement, as a workaround for
+		# MySQL >= 5.7 with sql_mode=only_full_group_by throwing ERROR 1055 (42000):
+		# Expression #1 of HAVING clause is not in GROUP BY clause
+		$t_sql = <<< ENDSQL
+SELECT lower(email) email, id, username
```

Comment:
Suspect that lower(email) must be removed to fix https://github.com/mantisbt/mantisbt/pull/1919/files#r1327981213